File: services/core-api/app/schemas/graph_context.py
"""Schemas for the graph context REST endpoint."""

from __future__ import annotations

from pydantic import BaseModel, Field


class RelatedStory(BaseModel):
    """A story related to the current one via graph connections."""

    id: str
    title: str
    snippet: str
    relevance: float


class EntityGroup(BaseModel):
    """Entities grouped by type."""

    people: list[dict[str, str]] = Field(default_factory=list)
    places: list[dict[str, str]] = Field(default_factory=list)
    events: list[dict[str, str]] = Field(default_factory=list)
    objects: list[dict[str, str]] = Field(default_factory=list)


class GraphContextResponse(BaseModel):
    """Response for GET /api/stories/{story_id}/graph-context."""

    related_stories: list[RelatedStory] = Field(default_factory=list)
    entities: EntityGroup = Field(default_factory=EntityGroup)

File: apps/web/src/features/evolve-workspace/api/graphContext.ts
import { apiGet } from '@/lib/api/client';

export interface RelatedStory {
  id: string;
  title: string;
  snippet: string;
  relevance: number;
}

export interface EntityGroup {
  people: Array<Record<string, string>>;
  places: Array<Record<string, string>>;
  events: Array<Record<string, string>>;
  objects: Array<Record<string, string>>;
}

export interface GraphContextResponse {
  related_stories: RelatedStory[];
  entities: EntityGroup;
}

export async function getGraphContext(storyId: string): Promise<GraphContextResponse> {
  return apiGet<GraphContextResponse>(`/api/stories/${storyId}/graph-context`);
}
